src/v/serde/protobuf/parser.cc

```diff
@@ -0,0 +1,546 @@
+                throw std::runtime_error(fmt::format(
+                  "legacy proto2 groups not supported (field={})",
+                  tag.field_number));
+            }
+            return current_->parser.bytes_left() > 0 ? ss::stop_iteration::no
+                                                     : ss::stop_iteration::yes;
+        });
+    }
+
+    void read_variant_field(int32_t field_number) {
+        auto* field_descriptor = current_->descriptor->FindFieldByNumber(
+          field_number);
+        if (field_descriptor == nullptr) {
+            std::ignore = read_varint<uint64_t>(current_->parser);
+            return;
+        }
+        switch (field_descriptor->type()) {
+        case google::protobuf::FieldDescriptor::TYPE_INT64:
+            update_field(
+              *field_descriptor,
+              read_varint<int64_t, zigzag_decode::no>(current_->parser));
+            break;
+        case google::protobuf::FieldDescriptor::TYPE_UINT64:
+            update_field(
+              *field_descriptor, read_varint<uint64_t>(current_->parser));
+            break;
+        case google::protobuf::FieldDescriptor::TYPE_INT32:
+            update_field(
+              *field_descriptor,
+              read_varint<int32_t, zigzag_decode::no>(current_->parser));
+            break;
+        case google::protobuf::FieldDescriptor::TYPE_SINT32:
+            update_field(
+              *field_descriptor, read_varint<int32_t>(current_->parser));
+            break;
+        case google::protobuf::FieldDescriptor::TYPE_SINT64:
+            update_field(
+              *field_descriptor, read_varint<int64_t>(current_->parser));
+            break;
+        case google::protobuf::FieldDescriptor::TYPE_BOOL:
+            update_field(
+              *field_descriptor,
+              static_cast<bool>(read_varint<int32_t>(current_->parser)));
+            break;
+        case google::protobuf::FieldDescriptor::TYPE_UINT32:
+            update_field(
+              *field_descriptor, read_varint<uint32_t>(current_->parser));
+            break;
+        case google::protobuf::FieldDescriptor::TYPE_ENUM:
+            update_field(
+              *field_descriptor,
+              static_cast<int32_t>(read_varint<uint64_t>(current_->parser)));
+            break;
+        case google::protobuf::FieldDescriptor::TYPE_FIXED64:
+        case google::protobuf::FieldDescriptor::TYPE_FIXED32:
+        case google::protobuf::FieldDescriptor::TYPE_SFIXED32:
+        case google::protobuf::FieldDescriptor::TYPE_SFIXED64:
+        case google::protobuf::FieldDescriptor::TYPE_DOUBLE:
+        case google::protobuf::FieldDescriptor::TYPE_FLOAT:
+        case google::protobuf::FieldDescriptor::TYPE_STRING:
+        case google::protobuf::FieldDescriptor::TYPE_GROUP:
+        case google::protobuf::FieldDescriptor::TYPE_MESSAGE:
+        case google::protobuf::FieldDescriptor::TYPE_BYTES:
+            // Skip invalid
```

Comment:
Should we throw here like we do in the other read_ methods?

Reply:
Amazingly, this is the behavior of the "normal" protobuf parser in the offical library. I have some tests to verify this - I'm not fully sure why this is the case, but compatibility for data corruption is probably desired.

Reply:
🤯 

yea i'm down for erring on the side of parity. Might be worth a small comment, given this reads like a bug

Reply:
Added a (probably longer than anyone wanted to read) comment